Refactor abstract job

ClementKunz opened the pull request:
The goal of this PR is to make the attribute `_exitInfo` from `AbstractJob` a private member. This change led to the modification of many methods' signatures to return an `ExitInfo` instead of setting the parameter `_exitInfo`.

ClementKunz committed:
refactor: runJob and canRun now return ExitInfo

ClementKunz committed:
refactor: make _exitInfo a private member

ClementKunz committed:
fix: fix log upload tests

ClementKunz committed:
fix: AbstractUploadSession

ClementKunz committed:
fix: fix integration tests on blacklist

ClementKunz committed:
fix: clean up blacklist after tests

luc-guyot-infomaniak commented on src/libsyncengine/jobs/network/abstractnetworkjob.cpp:
```diff
@@ -394,100 +393,87 @@ bool AbstractNetworkJob::receiveResponse(const Poco::URI &uri) {
-                        _exitInfo.code() != ExitCode::InvalidToken &&
-                        (_exitInfo.code() != ExitCode::BackError || _exitInfo.cause() != ExitCause::NotFound)) {
+                if (!exitInfo) {
+                    if (exitInfo.code() != ExitCode::Ok && exitInfo.code() != ExitCode::DataError &&
```
```suggestion
                    if (exitInfo.code() != ExitCode::DataError &&
```

ClementKunz committed:
fix: apply new logic to new gui jobs

ClementKunz committed:
fix: crash in error handling

luc-guyot-infomaniak commented on src/libsyncengine/jobs/network/kDrive_API/getsizejob.h:
```diff
@@ -31,12 +31,12 @@ class GetSizeJob : public AbstractTokenNetworkJob {
-        virtual std::string getSpecificUrl() override;
-        inline virtual ExitInfo setData() override { return ExitCode::Ok; }
+        std::string getSpecificUrl() override;
+        inline ExitInfo setData() override { return ExitCode::Ok; }
```
```suggestion
       ExitInfo setData() override { return ExitCode::Ok; }
```

luc-guyot-infomaniak commented on src/libsyncengine/update_detection/blacklist_changes_propagator/blacklistpropagator.cpp:
```diff
@@ -138,15 +134,15 @@ ExitCode BlacklistPropagator::removeItem(const NodeId &localNodeId, const NodeId
     }
 
-    SyncPath absolutePath = _sync.localPath() / localPath;
+    SyncPath absoluteLocalPath = _sync.localPath() / localPath;
```
```suggestion
    const SyncPath absoluteLocalPath = _sync.localPath() / localPath;
```

luc-guyot-infomaniak commented on src/libsyncengine/update_detection/blacklist_changes_propagator/blacklistpropagator.cpp:
```diff
@@ -213,16 +209,16 @@ ExitCode BlacklistPropagator::removeItem(const NodeId &localNodeId, const NodeId
+    if (!IoHelper::checkIfPathExists(absoluteLocalPath, exists, ioError)) {
         LOGW_WARN(Log::instance()->getLogger(),
-                  L"Error in IoHelper::checkIfPathExists for path=" << Utility::formatIoError(absolutePath, ioError));
+                  L"Error in IoHelper::checkIfPathExists for path=" << Utility::formatIoError(absoluteLocalPath, ioError));
```
```suggestion
                  L"Error in IoHelper::checkIfPathExists for " << Utility::formatIoError(absoluteLocalPath, ioError));
```

ClementKunz committed:
fix: fix tests

luc-guyot-infomaniak reviewed (commented):
General comment: replace `ExitInfo()` with `{}` or `ExitInfo{}` (C++11 guidline).

ClementKunz committed:
Apply suggestions from code review

Co-authored-by: Luc Guyot <162997198+luc-guyot-infomaniak@users.noreply.github.com>

ClementKunz committed:
Update src/libsyncengine/jobs/network/abstractnetworkjob.cpp

Co-authored-by: Luc Guyot <162997198+luc-guyot-infomaniak@users.noreply.github.com>

ClementKunz committed:
chore: address PR comment

ClementKunz commented:
> General comment: replace ExitInfo() with {} or ExitInfo{} (C++11 guidline).

Agreed, done in a82227ab488f37ef7f60fd8eb70a56d3ecb878f7

ClementKunz committed:
chore: Merge branch 'develop' into Refactor-AbstractJob

# Conflicts:
#	src/libsyncengine/jobs/network/abstractnetworkjob.cpp
#	src/libsyncengine/jobs/network/abstractnetworkjob.h
#	src/libsyncengine/jobs/network/abstracttokennetworkjob.cpp

ClementKunz committed:
chore: address Sonar's issue

ClementKunz committed:
chore: Merge branch 'develop' into Refactor-AbstractJob

# Conflicts:
#	src/server/comm/guijobs/loginrequesttokenjob.cpp
